### DISCUSSION TITLE
Password issues after password change via active directory / LDAP

### DISCUSSION THREAD

**Discussion by BeckeBauer** [kimai/kimai#4182] [On-Premise: Installation & Updates]
I use active directory / LDAP to authenticate when logging into Kimai. The set policy is a password change every 6 weeks. However, Kimai seems not to update the password after such a password change. Login fails:

```
request.INFO: Matched route "security_check". {"route":"security_check","route_parameters":{"_route":"security_check","_controller":"App\\Controller\\Security\\SecurityController::checkAction","_locale":"de"},"request_uri":"https://my-kimai/de/login_check","method":"POST"} []
security.INFO: Authenticator failed. {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\BadCredentialsException(code: 0): The presented password is invalid. at /kimai/vendor/symfony/security-http/EventListener/CheckCredentialsListener.php:69)","authenticator":"App\\Ldap\\LdapAuthenticator"} []
```
no matter if the previous password or the newly set password is used. Only after a "local" password update via `bin/console kimai:user:password ` to match the new password set with the active directory server, login works again.

Is there a way that Kimai updates the locally stored password if the password the user has entered matches the password from the LDAP server but not the locally stored password (if this is what the login prevents as the log indicates)?

**Comment by kevinpapst** [comment]
Sounds as if the LDAP setup is broken. 
Kimai will NEVER look in the local database, if the user is authenticated via LDAP.

**Reply by BeckeBauer** [reply]
I have configured LDAP and SAML in local.yaml. Could that play a role?

(Strange thing is that a test user added to the LDAP server and not existing in Kimai is added to Kimai without problems upon first login with the LDAP credentials - kind of contradicts the assumption that LDAP is broken - any other ideas how to test if LDAP works correctly?)

**Reply by BeckeBauer** [reply]
I can confirm that there is no (hashed) password in the db-table `password` under `kimai2_users` if user solely authenticates via LDAP. 
However, as soon as the users uses SAML, a password is stored and the db-table `auth` under `kimai2_users` is changed from LDAP to SAML. Then, using the regular login fields (username / password) to login causes my issue if the password is changed on the LDAP server. 

Is it necessary to store a (hashed) password if SAML is used? If so, could the password be automatically updated if there is a password change on the server?

Finally, is it possible that Kimai always checks which login is used (so that the login mechanism stored under `auth` will not interfere if login via username / password is used)?

**Reply by kevinpapst** [reply]
I don't get it.
You cannot run SAML and LDAP on the same time for the same user.
That is not supported.

**Reply by BeckeBauer** [reply]
Okay, I was not aware that I cannot use SAML and LDAP at the same time. This would explain my issues. 
It is a pity that this is not supported since both authentication measures seem complimentary and not contrary
